src/component/plotly-dash-preview/render.js

```diff
@@ -0,0 +1,65 @@
+const remote = require('../../util/remote')
+
+/**
+ * @param {object} info : info object
+ *  - url
+ *  - pdfOptions
+ * @param {object} opts : component options
+ * @param {function} sendToMain
+ *  - errorCode
+ *  - result
+ *    - imgData
+ */
+function render (info, opts, sendToMain) {
+  const result = {}
+
+  let win = remote.createBrowserWindow()
+  win.loadURL(info.url)
+
+  const contents = win.webContents
+
+  const done = errorCode => {
+    win.close()
+
+    if (errorCode) {
+      result.msg = 'dash preview generation failed'
+    }
+    sendToMain(errorCode, result)
+  }
+
+  /*
+   * We check for a 'waitfor' div in the dash-app
+   * which indicates that the app has finished rendering.
+   */
+  const finishedLoading = () => {
+    return win.webContents.executeJavaScript(`
+      new Promise((resolve, reject) => {
+        var a = document.getElementById("waitfor");
+        if (a) {
+          resolve(true);
+        }
+      })`)
+  }
+
+  win.on('closed', () => {
+    win = null
+  })
+
+  let intervalId = setInterval(() => {
```

Comment:
I'm curious. Why did you decide to place the `setInterval` in the main window scope instead of in the child window scope?

Would something like:

```js
const loaded = () => {
  return contents.executeJavaScript(`
     new Promise((resolve, reject) => {
       let tries = 10
       let interval = setInterval(() => {
         const el = document.getElementById('waitfor')
         if (el) {
           clearInterval(interval)
           resolve(true)
         }
         if (--tries === 0) reject('fail to load')
       }, 500)
     })
  `) 
}

loaded().then(() => {
   contents.printToPDF(info.pdfOptios, (err, pdfData) => {
      if (err) {
        done(525)
      } else {
        result.imgData = pdfData
        done()
      }
   }).catch(() => {
     done(525)
   })
})
```